spiketoolkit/postprocessing/utils.py

```diff
@@ -124,3 +124,78 @@ def get_max_channels_per_waveforms(recording, grouping_property, channel_ids, ma
         else:
             n_channels = max_channels_per_waveforms
     return n_channels
+
+
+def extract_snippet_from_traces(
+        traces,
+        start_frame,
+        end_frame,
+        channel_indices=None
+):
+    if (0 <= start_frame) and (end_frame <= traces.shape[1]):
+        x = traces[:, start_frame:end_frame]
+    else:
+        # handle edge cases
+        x = np.zeros((traces.shape[0], end_frame - start_frame), dtype=traces.dtype)
+        i1 = int(max(0, start_frame))
+        i2 = int(min(traces.shape[1], end_frame))
+        x[:, (i1 - start_frame):(i2 - start_frame)] = traces[:, i1:i2]
+    if channel_indices is not None:
+        x = x[channel_indices, :]
```

Comment:
Use `x = np.take(x, indices, axis=0)` you should have a small speedup.

Reply:
@samuelgarcia instead of what?

Reply:
x = x[channel_indices, :] is sugar syntax for x = np.take(x, indices, axis=0)
but calling directly is a slightly bit faster
if you do many many time (the case) here you could use the function (and maybe keep your line commented)

Reply:
@samuelgarcia that has been removed. Waveforms are extracted on all channels and trimmed later in case it's needed. Thanks anyways for the suggestion :)